### ISSUE TITLE
jwt is never updated

### ISSUE THREAD

**Issue by rasmusohrstig** [telenordigital/connect-android-sdk#100]
Once authenticated, the first jwt that you get is never updated on subsequent authentications. This means that we keep the same jwt forever (or until the app storage is cleared).

It looks like the bug could be fixed in ConnectStore.update(), but I have not tried this myself yet.

If you compare ConnectStore.update() with ConnectStore.set() it looks like there is some code missing in ConnectStore.update().

**Comment by jorunfa** [comment]
I think I understand the problem now, sorry we couldn't clarify this earlier. The ID token is only given by the server when exchanging the authorization code for access and refresh tokens request, not on any subsequent refresh calls. So you only get it once per user. That's why it's not updated on any later calls, because the intent/idea is to keep it around (even though it might be old), instead of clearing it. The iOS SDK actually does the opposite right now, where it clears the token on the first refresh, but as I understood it the request was to change that so we also never clear it on refresh.

@ulfcarlin, can you confirm, or have I misunderstood something?

**Comment by rasmusohrstig** [comment]
The jwt is actually not updated even when you log in with a different Connect ID, if you have not first logged out first. So you can end up with a jwt for Connect ID A and an access token for Connect ID B.

It might seem strange that we can log in with B without first logging out A, but it is possible. I will see if I can recreate this with some simple code.

**Comment by jorunfa** [comment]
I'm understanding more and more here - that sounds like a propper bug. But then I guess I would label that "it is possible to log in to a second user account without logging out first/without clearing the ID token".

**Comment by ulfcarlin** [comment]
@jorunfa: I think your understanding is correct. I suspect that only the ID token is handled incorrectly, though, while access and refresh tokens are handled correctly.

**Comment by rasmusohrstig** [comment]
Well, maybe.

My guess is that the bug could be solved by always updating the locally stored jwt when you get a new one from the Connect ID servers. I have not yet had the time to test this.

**Comment by jorunfa** [comment]
Hi again. I guess that is what we are doing, or at least attempting to do: you only get a new one when exchanging the authorization code for an access/refresh token pair.

> If a client requests the scope value openid, an ID token will also be returned when exchanging the authorization code for an access/refresh token pair.
http://docs.telenordigital.com/connect/id/id_token.html

This is where it's currently happening: https://github.com/telenordigital/connect-android-sdk/blob/master/connect/src/com/telenor/connect/id/ConnectIdService.java#L94

I tried to recreate the problem by making a branch for re-login, without logging out first, called `re-login-example`
https://github.com/telenordigital/connect-android-sdk/tree/re-login-example
https://github.com/telenordigital/connect-android-sdk/compare/re-login-example?expand=1

However, I was not able to recreate the problem. I added a button for refreshing the tokens as well, although this shouldn't change the ID token in any way, more for the sake of debugging.
